hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -812,7 +812,30 @@ func (r *NodePoolReconciler) reconcileMachineDeployment(log logr.Logger,
 
 	setMachineDeploymentReplicas(nodePool, machineDeployment)
 
+	// Bubble up AvailableReplicas and Ready condition from MachineDeployment.
```

Comment:
@enxebre this looks fine to me, but it does diverge from the pattern of reconciling all status at the beginning of the reconcile function. If that's ok, then lgtm

Reply:
Good point, should be fine for now, we can then consider move it up along with MHC condition.

The underlying condition contains invalid characters for metav1.Condition.Reason so I'm now looking into this, possibly need to create our own conditions type.

Reply:
Updated to move into our own condition struct.
In a follow up I will consider conflate some of the existing conditions and move dep/mhc up, drawback would be additional api server overhead.